docs/DeveloperGuide.md

```diff
@@ -1,34 +1,678 @@
+works in a similar manner for that of `Note` objects.
+
+![Figure 4.6-1](images/DeveloperGuide/Figure 4.6-1.png)
+_Figure 4.6-1: Sequence Diagram for creating and pushing old copies of Tasks_ 
+
+When the `undo` command is called, it retrieves the `editNumber` of the copied `Task` (or `Note`) at the top of the 
+stack in `taskHistory` (or `noteHistory`), and iterates through the existing `TaskList` (or `NoteList`) to see which 
+`Task` (or `Note`) has the corresponding `editNumber`. If there is a match, the existing `Task` is replaced with the 
+old copy, and then the old copy is removed from the `Tasklist`.
+
+The corresponding confirmation message to be displayed is determined by whether the `editNumber` is positive or negative.
+
+![Figure 4.6-2](images/DeveloperGuide/Figure 4.6-2.png)
+_Figure 4.6-2: Sequence Diagram for undoing edits or deletions of Tasks_
+
+##### Design Considerations
+
+To allow for multiple undos on the same `Task` (or `Note`) object, the `editNumber` of `Tasks` (or `Notes`) that have 
+been edited before must be checked. If it is anything but the default assigned _value(-999999)_, then its existing 
+`editNumber` will be taken and used as the `editID` for all successive copies made of it. This is to allow the 
+application to always find the same instance of the `Task` (or `Note`) inside the `TaskList` (or `NoteList`) when 
+restoring previous versions.
+
+### 4.7 Storing and Archiving Notes (Danzel)
+
+The storing and saving of data in the **LifEasier** app is done automatically after every change such as adding, editing, 
+deleting a component such as a lesson, deadline, event or note. The following section documents how the data storing 
+and archiving system of **LifEasier** was implemented, followed by the considerations taken during the design of the storage components.
+
+##### Implementation - Data saving and storing
+
+Figure 4.7-1 shows the simplified class diagram of all the components in the storage package. There are far more methods 
+that exist then as shown in the class diagram. These have been omitted for simplicity.
+
+![Class Diagram Image](images/DeveloperGuide/StorageClassDiagram.png)
+_Figure 4.7-1: Class Diagram for all storage components_
+
+Figure 4.7-2 shows the sequence diagram of the save data reading process which runs whenever **LifEasier** is run. Upon app startup, 
+the main `LifEasier` class creates a new `FileStorage` object, which starts the save reading process to load in all the previously stored 
+data of the user, if available. Else, new save directories and save files are created in the same directory which the `LifEasier.jar` was run. 
+Tasks and notes data read from the save file are used to create new `Task` and `Note` objects respectively, and added into `TaskList` and `NoteList`.
+
+![Startup file load sequence diagram](images/DeveloperGuide/StorageLaunchSequenceDiagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/20167103/97534584-36159c80-19f5-11eb-8a0b-2d1da9048490.png)

Should we removed the middle return" dotted lines? According to the standard notation, maybe there should not be any return in the middle of the activation bar.

The yellow highlighted dotted line maybe is misplaced.